Date: Wed, 24 Jan 2018 17:32:23 +0000
From: Leszek Swirski via gdb-patches <gdb-patches@sourceware.org>
To: gdb-patches@sourceware.org
Cc: Leszek Swirski <leszeks@google.com>
Subject: [PATCH] MI: Allow non-raw varobj evaluation

Make the MI variable object expression evaluation, with the
-var-evaluate-expression command, recursively call pretty printers, to
match the output of normal expression printing.

This is gated behind the -enable-pretty-printing command.

gdb/ChangeLog:

        * varobj.c (varobj_formatted_print_options): Allow recursive
        pretty printing if pretty printing is enabled.
---
 gdb/varobj.c  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdb/varobj.c b/gdb/varobj.c
index b6a2d8f369..f23243f3b7 100644
--- a/gdb/varobj.c
+++ b/gdb/varobj.c
@@ -2274,7 +2274,7 @@ varobj_formatted_print_options (struct value_print_options *opts,
 {
   get_formatted_print_options (opts, format_code[(int) format]);
   opts->deref_ref = 0;
-  opts->raw = 1;
+  opts->raw = !pretty_printing;
 }

 std::string
--
2.16.0.rc1.238.g530d649a79-goog
 


